src/peer-store/address-book.js

```diff
@@ -83,7 +189,10 @@ class AddressBook extends Book {
       }
     }
 
-    this._setData(peerId, addresses)
+    this._setData(peerId, {
+      addresses,
+      record: entry.record
+    })
```

Comment:
This is a bit concerning. Where is `set` being called? If we already have a signed peer record and `set` gets called it will override certified addresses. If we keep this behavior we **really** need to be clear in the docs about what set does. It would be pretty easy for a user to make a mistake and override signed addresses. I don't think we should prohibit this, but it's not a method that should be used regularly.

Reply:
> gets called it will override certified addresses.
Yes, that is true. Like map operations, set will replace what existed before.

In the libp2p codebase, `addressBook.set` was used in identify, which is not anymore as a result of identify using the signedPeerRecords now. It is used a lot in the tests.

I can add an attention notice to this in the `API.md` + jsdoc. I was thinking that we could deprecate set in the future. But, we still use `set` in ProtoBook and MetadataBook and we will use it as well.
If we decide to deprecate it, it should only be deprecated for the AddressBook, which would create a difference in their APIs